pkg/manager/member/tidb_init_manager.go

```diff
@@ -0,0 +1,381 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package member
+
+import (
+	"fmt"
+	"path"
+
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	apiequality "k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	batchlisters "k8s.io/client-go/listers/batch/v1"
+
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	"github.com/pingcap/tidb-operator/pkg/client/clientset/versioned"
+	"github.com/pingcap/tidb-operator/pkg/controller"
+	"github.com/pingcap/tidb-operator/pkg/label"
+)
+
+const (
+	initContainerName   = "wait"
+	containerName       = "mysql-client"
+	passwdKey           = "password"
+	passwdPath          = "/etc/tidb/password"
+	sqlKey              = "init-sql"
+	sqlPath             = "init.sql"
+	sqlDir              = "/data"
+	startScriptPath     = "start_script.py"
+	initStartScriptPath = "init_start_script.sh"
+	startScriptDir      = "/usr/local/bin"
+	startKey            = "start-script"
+	initStartKey        = "init-start-script"
+)
+
+// InitManager implements the logic for syncing TidbInitializer.
+type InitManager interface {
+	// Sync	implements the logic for syncing TidbInitializer.
+	Sync(*v1alpha1.TidbInitializer) error
+}
+
+type tidbInitManager struct {
+	jobLister  batchlisters.JobLister
+	jobControl controller.JobControlInterface
+	cmControl  controller.ConfigMapControlInterface
+	cli        versioned.Interface
+}
+
+// NewTiDBInitManager return tidbInitManager
+func NewTiDBInitManager(
+	jobLister batchlisters.JobLister,
+	jobControl controller.JobControlInterface,
+	cmControl controller.ConfigMapControlInterface,
+	cli versioned.Interface,
+) InitManager {
+	return &tidbInitManager{
+		jobLister,
+		jobControl,
+		cmControl,
+		cli,
+	}
+}
+
+func (tm *tidbInitManager) Sync(ti *v1alpha1.TidbInitializer) error {
+	if ti.DeletionTimestamp != nil {
+		return nil
+	}
```

Comment:
This logic should be placed [here](https://github.com/pingcap/tidb-operator/pull/1403/files#diff-5297cabf3cf775db17d396cac519f87aR154), make it fail fast.

Reply:
updated